File: software/examples/c/example_remote_callback.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_remote_switch_v2.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change XYZ to the UID of your Remote Switch Bricklet 2.0

// Callback function for remote status a callback
void cb_remote_status_a(uint8_t house_code, uint8_t receiver_code, uint8_t switch_to,
                        uint16_t repeats, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("House Code: %d\n", house_code);
	printf("Receiver Code: %d\n", receiver_code);
	printf("Switch To: %d\n", switch_to);
	printf("Repeats: %d\n", repeats);
	printf("\n");
}

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	RemoteSwitchV2 rs;
	remote_switch_v2_create(&rs, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Register remote status a callback to function cb_remote_status_a
	remote_switch_v2_register_callback(&rs,
	                                   REMOTE_SWITCH_V2_CALLBACK_REMOTE_STATUS_A,
	                                   (void *)cb_remote_status_a,
	                                   NULL);

	printf("Press key to exit\n");
	getchar();
	remote_switch_v2_destroy(&rs);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/php/ExampleRemoteCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletRemoteSwitchV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletRemoteSwitchV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Remote Switch Bricklet 2.0

// Callback function for remote status a callback
function cb_remoteStatusA($house_code, $receiver_code, $switch_to, $repeats)
{
    echo "House Code: $house_code\n";
    echo "Receiver Code: $receiver_code\n";
    echo "Switch To: $switch_to\n";
    echo "Repeats: $repeats\n";
    echo "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$rs = new BrickletRemoteSwitchV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Configure to receive from remote type A with minimum repeats set to 1 and enable callback
$rs->setRemoteConfiguration(BrickletRemoteSwitchV2::REMOTE_TYPE_A, 1, TRUE);

// Register remote status a callback to function cb_remoteStatusA
$rs->registerCallback(BrickletRemoteSwitchV2::CALLBACK_REMOTE_STATUS_A,
                      'cb_remoteStatusA');

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
